kernel/src/actions/mod.rs

```diff
@@ -152,6 +196,36 @@ impl Protocol {
             .as_ref()
             .is_some_and(|features| features.iter().any(|f| f == feature.as_ref()))
     }
+
+    /// Check if reading a table with this protocol is supported. That is: does the kernel support
+    /// the specified protocol reader version and all enabled reader features?
+    // TODO: should this return result so we can say _why_ it isn't supported?
+    pub fn is_read_supported(&self) -> bool {
+        if self.min_reader_version > KERNEL_READER_VERSION {
+            return false;
+        }
+        match &self.reader_features {
+            // check that all reader features are subset of supported
+            Some(reader_features) if self.min_reader_version == 3 => {
+                let parsed_features: Result<HashSet<_>, _> = reader_features
+                    .iter()
+                    .map(|s| s.parse::<ReaderFeatures>())
```

Comment:
Maybe this is cleaner?
```suggestion
                    .map(ReaderFeatures::from_str)
```

Reply:
Hm to do this I think I would have to add another `.map(String::as_str)` since we need to coerce &String to &str

Reply:
Then could also do:
```rust
.map(|x| ReaderFeatures::from_str(s))
```